src/python/pants/backend/visibility/rules.py

```diff
@@ -41,7 +43,12 @@ def build_file_visibility_implementation(
 @rule
 async def visibility_validate_dependencies(
     request: VisibilityValidateDependenciesRequest,
+    visibility: VisibilitySubsystem,
+    goals: CurrentExecutingGoals,
 ) -> ValidatedDependencies:
+    if not visibility.enforce or goals.is_running("lint"):
```

Comment:
Yay, finally. Getting this `goals.is_running("lint")` only took me from Nov to now to get here :p :D 
cc @AlexTereshenkov @stuhood

Reply:
Amazing! Will give a try this weekend - traveling at the moment